### PR TITLE
Add electron autoupdater

### PR DESCRIPTION
**Pull request by zvin**
Change-type: minor
Signed-off-by: Giovanni Garufi <giovanni@balena.io>

### CODE REVIEW

**Comment by thundron** [comment]
@resin-ci retest

**Comment by nazrhom** [comment]
test

File: lib/gui/etcher.js [hunk 1]
```diff
@@ -83,6 +85,25 @@ const createMainWindow = () => {
...
+    autoUpdater.on('update-available', (info) => {
+      console.log('update-available', info)
+    })
+    autoUpdater.on('update-not-available', (info) => {
+      console.log('update-not-available', info)
+    })
+    autoUpdater.on('error', (err) => {
+      console.log('error', err)
+    })
+    autoUpdater.checkForUpdatesAndNotify()
```
Comment (lurch) [inline lib/gui/etcher.js]: Should this also respect the 'updatesEnabled' setting?

Comment (thundron) [reply lib/gui/etcher.js]: Absolutely, which is why it's only a bunch of `console.log`s for now

**Comment by thundron** [comment]
Closing and opening a new one to get correct reviews

**Pull request closed without merging**